enhancements/rhcos/split-rhcos-into-layers.md

```diff
@@ -0,0 +1,458 @@
+no longer be versioned with OpenShift. (But note that the format itself remains
+the same, and is still semver-compatible.)
+
+This change is likely to throw off anything that currently parses the version
+string to for example derive the OCP and RHEL versions for which an RHCOS image
+was built. Those will need to be adapted if they deal with bootimages. The
+release browser as mentioned above is a likely candidate.
+
+Additionally, RHCOS artifacts on the official mirrors are currently versioned
+by the OCP release (e.g. `rhcos-4.16.3-x86_64-live.x86_64.iso`) and kept under
+OCP-versioned directories (e.g. `4.16.3/`). We don't want customers to have to
+know which version of RHCOS bootimages to download for their target OCP version.
+So to avoid confusing customers, we should still have OCP-versioned directories
+but it would be wasteful to not symlink the bootimages on the server side to
+share across OCP versions that share the same set of bootimages.
+
+As such, the versioning change should be transparent to end-users, but on the
+off chance that a customer is relying on the current semantics, we should still
+announce it in the GA release notes.
+
+#### Hidden dependencies
+
+Given that we're effectively removing packages from the bootimages, we may
+uncover hidden dependencies on the removed packages both within other components
+of OpenShift and in end-user code.
+
+Regarding other OCP components, the goal is to catch these dependencies during
+CI testing of the installer PR that switches over to the new bootimages. We
+need to raise awareness of the change to the whole OCP org and call for extra
+testing. This needs to be a coordinated event with ART, DPTP, and TRT. Of
+course, even after merging, if any regression is found, we can easily revert the
+PR to get back to green.
+
+Regarding end-user code, although bootimages are by their very nature
+transient (we boot, and shortly after pivot into the node image), there are
+some situations where user-provided code may run (e.g. when manually extending
+`bootstrap.ign` or `master.ign` or when scripting the live ISO). This code may
+be relying on packages that are going away (e.g. some local-only `oc` command,
+or grepping `--help` texts of the kubelet, etc...).
+
+We shouldn't worry *too much* about breaking these use cases, but it still seems
+worth highlighting in the GA release notes that these OCP packages are no longer
+present.
+
+### Drawbacks
+
+- This will make it somewhat harder (or at the very least much more explicit) to
+  fast-track RHEL fixes into specific OCP releases. Because everything beneath
+  the node layer is RHEL-versioned and possibly shared across multiple OCP
+  releases, any fast-tracks would need to happen by doing an RPM upgrade within
+  the node layer. This is more cumbersome than just tagging packages into the
+  the ART plashets as currently done, but makes us more thorough in our known
```

Comment:
Can you unpack "cumbersome"?
This seems vital to deliver a hotfix. I can understand it not being ideal from an image size perspective, but still possible, right?